src/System.Private.CoreLib/shared/System/SpanHelpers.Byte.cs

```diff
@@ -270,6 +270,29 @@ public static unsafe int IndexOf(ref byte searchSpace, byte value, int length)
             {
                 if ((int)(byte*)offset < length)
                 {
+                    if ((((nuint)Unsafe.AsPointer(ref searchSpace) + (nuint)offset) & (nuint)(Vector256<byte>.Count - 1)) != 0)
```

Comment:
Does this also need `if ((byte*)nLength > (byte*)offset)` check to make sure that there is enough bytes in the buffer?

Reply:
No, this is just after sequential processing, if its got here either there is 0 left or >= Vector length.

Probably could use a comment to that effect though.

Reply:
Added comment